graph/api/src/main/java/org/jboss/windup/graph/service/ProjectService.java

```diff
@@ -67,4 +78,62 @@ public ProjectModel getOrCreateSharedLibsProject()
 
         return sharedLibsProject;
     }
+
+    public Map<ProjectModel, ProjectModel> getProjectToRootProjectMap()
+    {
+        Map<ProjectModel, ProjectModel> projectModels = new HashMap<>();
+
+        for (FileModel inputPath : WindupConfigurationService.getConfigurationModel(this.getGraphContext()).getInputPaths())
+        {
+            ProjectModel rootProjectModel = inputPath.getProjectModel();
+            if (rootProjectModel == null)
+            {
+                continue;
+            }
+
+            ProjectModelTraversal traversal = new ProjectModelTraversal(rootProjectModel, new OnlyOnceTraversalStrategy());
```

Comment:
This should probably be the All strategy rather than "OnlyOnce", if I have read the intent correctly here.